### PR TITLE
ENT-4494 Harmonize Kryo serialalization 

### PR DESCRIPTION
**Pull request by blsemo**
Move custom serializer for iterators/collections to OS. Added Rick as it might have a performance impact

### CODE REVIEW

**Review by fowlerrr — DISMISSED** [review]
Everything I've touched looks okay. Can't really comment on the SGX bit or the `SNAPPY`.

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/kryo/KryoCheckpointSerializer.kt [hunk 1]
```diff
@@ -118,6 +120,6 @@ val KRYO_CHECKPOINT_CONTEXT = CheckpointSerializationContextImpl(
         QuasarWhitelist,
         emptyMap(),
         true,
-        null,
+        SNAPPY,
```
Comment (rick-r3) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/kryo/KryoCheckpointSerializer.kt]: I thought the idea was to default to no SNAPPY ?

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/kryo/KryoCheckpointSerializer.kt [hunk 2]
```diff
@@ -15,6 +15,8 @@ import net.corda.core.serialization.internal.CheckpointSerializationContext
 import net.corda.core.serialization.internal.CheckpointSerializer
 import net.corda.core.utilities.ByteSequence
 import net.corda.serialization.internal.*
+import net.corda.serialization.internal.CordaSerializationEncoding.SNAPPY
```
Comment (rick-r3) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/kryo/KryoCheckpointSerializer.kt]: Able to clean up these imports?

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/kryo/KryoCheckpointSerializer.kt [hunk 3]
```diff
@@ -15,6 +15,8 @@ import net.corda.core.serialization.internal.CheckpointSerializationContext
...
+import java.security.PublicKey
```
Comment (mnesbit) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/serialization/kryo/KryoCheckpointSerializer.kt]: Not needed

**Review by mnesbit — APPROVED** [review]
LGTM

**Review by fowlerrr — APPROVED** [review]
Yep, all my changes still look okay (everything but the SGX/"by lazy" stuff)

**Review by rick-r3 — APPROVED** [review]
LGTM

**Comment by rick-r3** [comment]
Do you want me to hold off from merging until the enterprise complement is green?

**Pull request merged**